lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -899,35 +899,113 @@ class MongoClientInterface {
             return cb(errors.InternalError);
         });
     }
-
     /**
-     * Put object when versioning is not enabled
-     * @param {Object} c bucket collection
-     * @param {String} bucketName bucket name
-     * @param {String} objName object name
-     * @param {Object} objVal object metadata
-     * @param {Object} params params
-     * @param {Object} log logger
-     * @param {Function} cb callback
-     * @return {undefined}
+     * Puts an object into a MongoDB collection. Depending on the parameters,
+     * the object is either directly put into the collection
+     * or the existing object is marked as deleted and a new object is inserted.
+     *
+     * @param {Object} collection - The MongoDB collection to put the object into.
+     * @param {string} bucketName - The name of the bucket the object belongs to.
+     * @param {string} objName - The name of the object.
+     * @param {Object} value - The value of the object.
+     * @param {Object} params - Additional parameters. If params.doesNotNeedOpogUpdate is true, the object is directly
+     * put into the collection. If params.vFormat is provided, it is used to format the object key.
+     * @param {Object} log - The logger to use.
+     * @param {Function} cb - The callback function to call when the operation is complete. It is called with an error
+     * if there is an issue with the operation.
+     * @param {string} [originOp='s3:LifecycleTransitionPut'] - The operation that originated this action.
+     * @returns {undefined}
      */
-    putObjectNoVer(c, bucketName, objName, objVal, params, log, cb) {
-        const masterKey = formatMasterKey(objName, params.vFormat);
-        c.updateOne({
-            _id: masterKey,
-        }, {
-            $set: {
-                _id: masterKey,
-                value: objVal,
-            },
-        }, {
-            upsert: true,
-        }).then(() => cb()).catch((err) => {
-            log.error('putObjectNoVer: error putting obect with no versioning', { error: err.message });
-            return cb(errors.InternalError);
+    putObjectNoVer(collection, bucketName, objName, value, params, log, cb, originOp = 's3:LifecycleTransitionPut') {
```

Comment:
why would it have such a default?
- this is an abstract metadata interface, there should not be any default related to a specific use case.
- in the scenario we are trying to solve, there is no "lifecycle" on this specific (user!) operation: this should be more something like `ReplaceArchivedObject`